src/Motions/Document.ts

```diff
@@ -31,11 +39,14 @@ export class MotionDocument extends Motion {
 
         const document = activeTextEditor.document;
 
-        let line = this.line;
-        line = Math.max(0, this.line);
-        line = Math.min(document.lineCount, this.line);
-
-        return from.with(line);
+        let line = (this.percent === undefined)
+            ? this.line
+            : Math.round((document.lineCount-1) * this.percent);
+        
+        line = Math.max(0, line);
+        line = Math.min(document.lineCount - 1, line);
+        let lineText = document.lineAt(line);
```

Comment:
Since we are not changing the value of `lineText`, `const` is better than `let`.
Adding empty line before and after this assignment will make the code more readable too 😄